src/main/java/kr/syeyoung/dungeonsguide/dungeon/doorfinder/DungeonSpecificDataProvider.java

```diff
@@ -18,22 +18,57 @@
 
 package kr.syeyoung.dungeonsguide.dungeon.doorfinder;
 
-import kr.syeyoung.dungeonsguide.roomprocessor.RoomProcessor;
-import kr.syeyoung.dungeonsguide.roomprocessor.bossfight.BossfightProcessor;
+import com.google.common.collect.Sets;
+import kr.syeyoung.dungeonsguide.dungeon.roomprocessor.bossfight.BossfightProcessor;
+import net.minecraft.entity.item.EntityArmorStand;
+import net.minecraft.init.Blocks;
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.World;
 
 import javax.vecmath.Vector2d;
+import java.util.Collection;
+import java.util.Set;
 
-public interface DungeonSpecificDataProvider {
-    BlockPos findDoor(World w, String dungeonName);
-    Vector2d findDoorOffset(World w, String dungeonName);
+public abstract class DungeonSpecificDataProvider {
 
-    BossfightProcessor createBossfightProcessor(World w, String dungeonName);
+    private static final Set<Vector2d> directions = Sets.newHashSet(new Vector2d(0,1), new Vector2d(0, -1), new Vector2d(1, 0), new Vector2d(-1 , 0));
 
-    boolean isTrapSpawn(String dungeonName);
+    public BlockPos findDoor(World w, String dungeonName) {
```

Comment:
This only works for catacombs, not future dungeons. Send it back to CatacombDataProvider / MasterDataProvider

Reply:
the npc's name might as well not be mort!

Reply:
I did not mean you to create MortDetector2000

findDoor is Dungeon Specific
that is, if hypixel changes the start room's, we will have to modify it to fit our needs

AND that is why mort should not be in abstract class

Revert it back to CatacombsProvier

Reply:
Actually just make the Provider really abstract and we're good to go

Reply:
Please